### DISCUSSION TITLE
What is the equivalent for MqttClientOptionsBuilder extension `WithTrustChain` ?

### DISCUSSION THREAD

**Discussion by weirdyang** [dotnet/MQTTnet#2047] [Q&A]
For .NET 6 and below, how do I reproduce the extension method `WithTrustChain` ?

```
#if NET7_0_OR_GREATER
        public MqttClientTlsOptionsBuilder WithTrustChain(X509Certificate2Collection chain)
        {
            _tlsOptions.TrustChain = chain;
            return this;
        }

#endif
```

**Comment by weirdyang** [comment]
I copied the code from here and it seems to work:
https://github.com/iotmodels/MQTTnet.Extensions.MultiCloud/blob/master/src/MQTTnet.Extensions.MultiCloud/Connections/X509ChainValidator.cs

**Reply by rido-min** [reply]
I'd say this is a better version for X509ChainValidator

https://github.com/Azure-Samples/MqttApplicationSamples/blob/701367c9de7145362a85c36747673822cead8af0/mqttclients/dotnet/MQTTnet.Client.Extensions/X509ChainValidator.cs

**Reply by weirdyang** [reply]
Thanks will update to use that, btw I noticed you maintain both of those repos, do you have any recommendations on how to better my understanding of certificate validation etc? or perhaps how did you learn what is required?
